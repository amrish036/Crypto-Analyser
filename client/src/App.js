import React, { Component } from 'react';
import Typography from '@material-ui/core/Typography';
import CssBaseline from '@material-ui/core/CssBaseline';
import Button from '@material-ui/core/Button';
import Currency from './Models/Currency';
import CurrencyQuoteCard from './Components/CurrencyQuoteCard';
import axios from 'axios';
import './Styles/App.css';
import FilterFormComponent from './Components/FilterFormComponent.jsx';
import AppBarComponent from './Components/AppBarComponent';
import FooterComponent from './Components/FooterComponent';


const ButtonStyle = {
  margin: '20px',
}

var data = []

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      items: null,
      currencies: [Currency],
      open: false,
      quotesFound: false,
    }
    this.selectedDate = null;
    this.selectedCurrency = null;
    this.onRefreshButtonClick = this.onRefreshButtonClick.bind(this)

  }

  //Compnent lifecycle events

  componentWillMount() {
    this.getDatafromDB();
  }

  componentDidMount() {
    this.setState({
      items: null,
      currencies: data
    })
  }

  // Handling events

  handleClickOpen = () => {
    this.selectedCurrency = null;
    this.selectedDate = null;
    this.setState({ open: true });
  };

  onRefreshButtonClick() {
    this.getDatafromDB();
  }

  getFormData(open, date, currency) {
    this.setState({ open: open });

    if (currency) {
      this.selectedCurrency = currency;
    }
    if (date) {
      this.selectedDate = date;
    }
    if (currency || date) { this.fetchNewDataFromDb(); }

  }

  //API calls

  async getDatafromDB() {
    try {
      return await axios.get("/api/getData")
        .then(response => {
          this.setState({ items: response.data.data })
        })
    } catch (error) {
      console.log(error)
    }
  };

  fetchNewDataFromDb() {
    try {
      axios.get("/api/fetchNewData", {
        params: {
          date: this.selectedDate,
          currency: this.selectedCurrency
        }
      })
        .then(response => {
          this.setState({
            items: response.data.data,
          })
        })
      { console.log("Fetched  new datafrom DB") }
    }
    catch (error) {
      console.log(error)
    }
  }

  render() {
    var Currencies = [];
    if (this.state.items != null) {
      data = this.state.items;
      Currencies = data;
    }
    else {
      Currencies = data;
    }

    return (
      <React.Fragment>
        <CssBaseline />
        <div className="App" >
          <AppBarComponent />
          <header className="App-header" >
            <div className="CardContainer">
              {Currencies.length ? Currencies.map(currency => <CurrencyQuoteCard quotes={currency} key={currency.id} />)
                : <Typography variant="h6" color="inherit" noWrap>
                  No Quote Found, try another Date or Currency
                </Typography>}
            </div>
            <div>
              <Button className="refreshButton" variant="contained" color="primary" onClick={this.onRefreshButtonClick} style={ButtonStyle}>
                Refresh Quotes
              </Button>
              <Button className="filterButton" variant="contained" color="primary" style={ButtonStyle} onClick={this.handleClickOpen}>Filter</Button>
              {this.state.open ? <FilterFormComponent sendData={this.getFormData.bind(this)} /> : <></>}
            </div>
          </header>
          <FooterComponent />
        </div>
      </React.Fragment>
    );
  }
}

export default App;
