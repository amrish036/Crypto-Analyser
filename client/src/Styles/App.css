.App {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
}
.App-header {
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

footer{
  background-color: #282c34;
  margin-top: 10px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
}

.CardContainer{
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  flex-wrap: wrap;
}


