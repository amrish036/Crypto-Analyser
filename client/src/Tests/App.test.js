import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { createShallow } from '@material-ui/core/test-utils';


configure({ adapter: new Adapter() });

describe('<App />', () => {

  let wrapper =  beforeEach(()=>{ wrapper = shallow(<App />) });
  
  it('includes one app component', () =>{
    expect(wrapper.find('div.App')).toHaveLength(1);
  });
  
  it('includes text in component', () =>{
    expect('Quote').toMatch('Quote');
  });
  
  it('includes one refresh button', () =>{
    expect(wrapper.find('.refreshButton')).toHaveLength(1);
  });
  
  it('includes one filter button', () =>{
    expect(wrapper.find('.refreshButton')).toHaveLength(1);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the App component snapshot',()=>{
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

})

