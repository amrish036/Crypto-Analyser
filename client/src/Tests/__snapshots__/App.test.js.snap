// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the App component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(CssBaseline) />,
        <div
          className="App"
        >
          <AppBarComponent />
          <header
            className="App-header"
          >
            <div
              className="CardContainer"
            >
              <WithStyles(Typography)
                color="inherit"
                noWrap={true}
                variant="h6"
              >
                No Quote Found, try another Date or Currency
              </WithStyles(Typography)>
            </div>
            <div>
              <WithStyles(Button)
                className="refreshButton"
                color="primary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": "20px",
                  }
                }
                variant="contained"
              >
                Refresh Quotes
              </WithStyles(Button)>
              <WithStyles(Button)
                className="filterButton"
                color="primary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": "20px",
                  }
                }
                variant="contained"
              >
                Filter
              </WithStyles(Button)>
              <React.Fragment />
            </div>
          </header>
          <FooterComponent />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AppBarComponent />,
            <header
              className="App-header"
            >
              <div
                className="CardContainer"
              >
                <WithStyles(Typography)
                  color="inherit"
                  noWrap={true}
                  variant="h6"
                >
                  No Quote Found, try another Date or Currency
                </WithStyles(Typography)>
              </div>
              <div>
                <WithStyles(Button)
                  className="refreshButton"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "20px",
                    }
                  }
                  variant="contained"
                >
                  Refresh Quotes
                </WithStyles(Button)>
                <WithStyles(Button)
                  className="filterButton"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "20px",
                    }
                  }
                  variant="contained"
                >
                  Filter
                </WithStyles(Button)>
                <React.Fragment />
              </div>
            </header>,
            <FooterComponent />,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="CardContainer"
                >
                  <WithStyles(Typography)
                    color="inherit"
                    noWrap={true}
                    variant="h6"
                  >
                    No Quote Found, try another Date or Currency
                  </WithStyles(Typography)>
                </div>,
                <div>
                  <WithStyles(Button)
                    className="refreshButton"
                    color="primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "margin": "20px",
                      }
                    }
                    variant="contained"
                  >
                    Refresh Quotes
                  </WithStyles(Button)>
                  <WithStyles(Button)
                    className="filterButton"
                    color="primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "margin": "20px",
                      }
                    }
                    variant="contained"
                  >
                    Filter
                  </WithStyles(Button)>
                  <React.Fragment />
                </div>,
              ],
              "className": "App-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <WithStyles(Typography)
                    color="inherit"
                    noWrap={true}
                    variant="h6"
                  >
                    No Quote Found, try another Date or Currency
                  </WithStyles(Typography)>,
                  "className": "CardContainer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "No Quote Found, try another Date or Currency",
                    "color": "inherit",
                    "noWrap": true,
                    "variant": "h6",
                  },
                  "ref": null,
                  "rendered": "No Quote Found, try another Date or Currency",
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <WithStyles(Button)
                      className="refreshButton"
                      color="primary"
                      onClick={[Function]}
                      style={
                        Object {
                          "margin": "20px",
                        }
                      }
                      variant="contained"
                    >
                      Refresh Quotes
                    </WithStyles(Button)>,
                    <WithStyles(Button)
                      className="filterButton"
                      color="primary"
                      onClick={[Function]}
                      style={
                        Object {
                          "margin": "20px",
                        }
                      }
                      variant="contained"
                    >
                      Filter
                    </WithStyles(Button)>,
                    <React.Fragment />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Refresh Quotes",
                      "className": "refreshButton",
                      "color": "primary",
                      "onClick": [Function],
                      "style": Object {
                        "margin": "20px",
                      },
                      "variant": "contained",
                    },
                    "ref": null,
                    "rendered": "Refresh Quotes",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Filter",
                      "className": "filterButton",
                      "color": "primary",
                      "onClick": [Function],
                      "style": Object {
                        "margin": "20px",
                      },
                      "variant": "contained",
                    },
                    "ref": null,
                    "rendered": "Filter",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Symbol(react.fragment),
                  },
                ],
                "type": "div",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(CssBaseline) />,
          <div
            className="App"
          >
            <AppBarComponent />
            <header
              className="App-header"
            >
              <div
                className="CardContainer"
              >
                <WithStyles(Typography)
                  color="inherit"
                  noWrap={true}
                  variant="h6"
                >
                  No Quote Found, try another Date or Currency
                </WithStyles(Typography)>
              </div>
              <div>
                <WithStyles(Button)
                  className="refreshButton"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "20px",
                    }
                  }
                  variant="contained"
                >
                  Refresh Quotes
                </WithStyles(Button)>
                <WithStyles(Button)
                  className="filterButton"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "20px",
                    }
                  }
                  variant="contained"
                >
                  Filter
                </WithStyles(Button)>
                <React.Fragment />
              </div>
            </header>
            <FooterComponent />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <AppBarComponent />,
              <header
                className="App-header"
              >
                <div
                  className="CardContainer"
                >
                  <WithStyles(Typography)
                    color="inherit"
                    noWrap={true}
                    variant="h6"
                  >
                    No Quote Found, try another Date or Currency
                  </WithStyles(Typography)>
                </div>
                <div>
                  <WithStyles(Button)
                    className="refreshButton"
                    color="primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "margin": "20px",
                      }
                    }
                    variant="contained"
                  >
                    Refresh Quotes
                  </WithStyles(Button)>
                  <WithStyles(Button)
                    className="filterButton"
                    color="primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "margin": "20px",
                      }
                    }
                    variant="contained"
                  >
                    Filter
                  </WithStyles(Button)>
                  <React.Fragment />
                </div>
              </header>,
              <FooterComponent />,
            ],
            "className": "App",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="CardContainer"
                  >
                    <WithStyles(Typography)
                      color="inherit"
                      noWrap={true}
                      variant="h6"
                    >
                      No Quote Found, try another Date or Currency
                    </WithStyles(Typography)>
                  </div>,
                  <div>
                    <WithStyles(Button)
                      className="refreshButton"
                      color="primary"
                      onClick={[Function]}
                      style={
                        Object {
                          "margin": "20px",
                        }
                      }
                      variant="contained"
                    >
                      Refresh Quotes
                    </WithStyles(Button)>
                    <WithStyles(Button)
                      className="filterButton"
                      color="primary"
                      onClick={[Function]}
                      style={
                        Object {
                          "margin": "20px",
                        }
                      }
                      variant="contained"
                    >
                      Filter
                    </WithStyles(Button)>
                    <React.Fragment />
                  </div>,
                ],
                "className": "App-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <WithStyles(Typography)
                      color="inherit"
                      noWrap={true}
                      variant="h6"
                    >
                      No Quote Found, try another Date or Currency
                    </WithStyles(Typography)>,
                    "className": "CardContainer",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "No Quote Found, try another Date or Currency",
                      "color": "inherit",
                      "noWrap": true,
                      "variant": "h6",
                    },
                    "ref": null,
                    "rendered": "No Quote Found, try another Date or Currency",
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <WithStyles(Button)
                        className="refreshButton"
                        color="primary"
                        onClick={[Function]}
                        style={
                          Object {
                            "margin": "20px",
                          }
                        }
                        variant="contained"
                      >
                        Refresh Quotes
                      </WithStyles(Button)>,
                      <WithStyles(Button)
                        className="filterButton"
                        color="primary"
                        onClick={[Function]}
                        style={
                          Object {
                            "margin": "20px",
                          }
                        }
                        variant="contained"
                      >
                        Filter
                      </WithStyles(Button)>,
                      <React.Fragment />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Refresh Quotes",
                        "className": "refreshButton",
                        "color": "primary",
                        "onClick": [Function],
                        "style": Object {
                          "margin": "20px",
                        },
                        "variant": "contained",
                      },
                      "ref": null,
                      "rendered": "Refresh Quotes",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Filter",
                        "className": "filterButton",
                        "color": "primary",
                        "onClick": [Function],
                        "style": Object {
                          "margin": "20px",
                        },
                        "variant": "contained",
                      },
                      "ref": null,
                      "rendered": "Filter",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Symbol(react.fragment),
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
